src/Alfred/Cache.hs

```diff
@@ -1,11 +1,34 @@
 module Alfred.Cache
     ( writeItems
+    , rawWriteItems
     ) where
 
 import qualified Data.Csv as Csv
 import qualified Data.ByteString.Lazy as BS
 
 import Alfred.Item
+import Alfred.Search
+import qualified Data.HashMap.Strict as HM
 
 writeItems :: String -> [Item] -> IO ()
-writeItems cacheFile = BS.writeFile cacheFile . Csv.encodeDefaultOrderedByName
+writeItems cacheFile items = do
+  cachedItems <- getItems cacheFile
+  rawWriteItems cacheFile $ itemsWithCachedVisits items cachedItems
+
+rawWriteItems :: String -> [Item] -> IO ()
```

Comment:
I don't think so; `rawWriteItems` is used internally only by `Increment.hs` to update specific items with a new visit count.